src/screens/Exchange/ExchangeConfirm.js

```diff
@@ -432,102 +350,76 @@ class ExchangeConfirmScreen extends React.Component<Props, State> {
     });
     const formattedReceiveAmount = formatAmountDisplay(receiveQuantity);
 
+    const receiveAmountInFiat = parseFloat(receiveQuantity) * getRate(rates, toAssetCode, fiatCurrency);
+    const formattedReceiveAmountInFiat = formatFiat(receiveAmountInFiat, fiatCurrency);
+
     const providerLogo = getOfferProviderLogo(provider, theme, 'vertical');
-    const confirmButtonTitleDefault = setTokenAllowance ? t('exchangeContent.button.enableAsset') : t('button.confirm');
-    const confirmButtonTitle = gettingFee ? t('label.gettingFee') : confirmButtonTitleDefault;
+    const confirmButtonTitle = gettingFee ? t('label.gettingFee') : t('button.confirm');
 
     return (
       <ContainerWithHeader
         headerProps={{
           centerItems: [{ title: t('exchangeContent.title.confirmScreen') }],
           customOnBack: this.handleBack,
         }}
+        inset={{ bottom: 'never' }}
       >
         <ScrollWrapper contentContainerStyle={{ minHeight: '100%' }}>
           <MainWrapper>
-            {!setTokenAllowance &&
-              <ExchangeScheme
-                fromValue={payQuantity}
-                fromAssetCode={fromAssetCode}
-                toValue={formattedReceiveAmount}
-                toAssetCode={toAssetCode}
-                imageSource={providerLogo}
+            <ExchangeScheme
+              fromValue={payQuantity}
+              fromAssetCode={fromAssetCode}
+              toValue={formattedReceiveAmount}
+              toValueInFiat={formattedReceiveAmountInFiat}
+              toAssetCode={toAssetCode}
+              imageSource={providerLogo}
+            />
+            <Spacing h={36} />
+            <TableWrapper>
+              <Table title={t('exchangeContent.label.exchangeDetails')}>
+                <TableRow>
+                  <TableLabel>{t('exchangeContent.label.exchangeRate')}</TableLabel>
+                  <Row>
+                    <ExchangeIcon name="exchange" />
+                    <Spacing w={4} />
+                    <BaseText regular>
+                      {t('exchangeContent.label.exchangeRateLayout', {
+                        rate: (parseFloat(receiveQuantity) / parseFloat(payQuantity)).toPrecision(2),
+                        toAssetCode,
+                        fromAssetCode,
+                      })}
+                    </BaseText>
+                  </Row>
+                </TableRow>
+                <TableRow>
+                  <TableLabel>{t('exchangeContent.label.maxSlippage')}</TableLabel>
+                  <BaseText regular primary> {t('percentValue', { value: ALLOWED_SLIPPAGE })}</BaseText>
+                </TableRow>
+              </Table>
+              <Spacing h={20} />
+              <Table title={t('transactions.label.fees')}>
```

Comment:
As a value we can have a fee, an amount, an exchange rate, a date, a time, an user and who knows what in the future. So I imagine we'd have an array with objects like this (simplified):
```
[
 {
   label: "Ethereum fee",
   value: {
     type: "fee",
     txFeeInWei: txFeeInWei,
     gasToken: gasToken,
  },
 },
...
]
```
And in the Table component we should iterate over objects and render them depending on the type of the value. So we'd have a long switch block. And every time we'd need something different on new screen (for example a weird icon just on this screen - see sablier time icon) we'd need to modify this long switch block just for this screen. And we end up with the next ListItemWithImage or ActivityFeedItem.
Instead we now use composition where every screen can render in the table whatever it wants without modifying the Table component. If a component is used frequently (a TableFee component for example) it's extracted as separate component to not repeat it's logic. + we have type checking nearly for free, we just need to write props and that's easy.
And if we look at Table creation now it looks quite similar to the objects approach. It's just more xml-ish instead of json but even the amount of text I need to type is similar. And if you don't like importing objects every time (however I've taken care so it's all just in one import) you can do something like`import * from 'components/Table`.

Reply:
I've imagined it more like 

```
[
 {
   label: "Ethereum fee",
   value: <SomeComponent />,
 },
...
]
```

But yeah, I understand your concern :)) I'm good with it staying as it is :)